cmd2/cmd2.py

```diff
@@ -2271,15 +2241,13 @@ def _cmdloop(self) -> bool:
             # Enable tab completion
             readline.parse_and_bind(self.completekey + ": complete")
 
-        stop = False
         try:
+            stop = False
             while not stop:
                 if self.cmdqueue:
-                    # Run command out of cmdqueue if nonempty (populated by load command or commands at invocation)
-                    line = self.cmdqueue.pop(0)
-
-                    if self.echo and line != 'eos':
-                        self.poutput('{}{}'.format(self.prompt, line))
+                    # Run commands out of cmdqueue if nonempty (populated by commands at invocation)
+                    stop = self.runcmds_plus_hooks(self.cmdqueue)
+                    self.cmdqueue.clear()
```

Comment:
I'm not sure I like this implementation relative to the old one where we just pop an element off the queue. 

What if the act of running commands in the queue actually adds commands to the queue?  Or what if there is a background thread running which adds commands to the queue?  This would then immediately delete those commands.  

I strongly recommend you switch this back to an implementation that pops a single command off the queue.

Reply:
Agree with @tleonhardt, we should go back to the implementation that pops a single command off the queue when we need a new command. As coded in this PR, the behavior of running the script could be different that what happened when it first happened. For example, if the third command in the script adds a command to `cmdqueue`, that command won't get executed when we run the script.